build.sh

```diff
@@ -32,60 +32,76 @@ region = $HEROKLES_AWS_REGION
 eocre
 
 echo "Getting environment variables."
-JSON=
-JSON_FULL=$( aws ssm get-parameters --name /${PROJECT}/${ENV} )
-
-if [[ ! -z $( echo "$JSON_FULL" | jq -r '.InvalidParameters | .[]' ) ]] ; then
-  if [[ $ENV == pr-${PR_NUM:-''} ]] ; then
-    echo "New PR deployment, copying env vars from the template."
-    JSON=$( aws ssm get-parameters --name /${PROJECT}/prs | jq -r '.Parameters | .[] | .Value' )
+JSON=$( mktemp )
+JSON_TEMPLATE=$( mktemp )
+JSON_FULL=$( mktemp )
+aws ssm get-parameters --name /${PROJECT}/${ENV} > $JSON_FULL
+
+if [[ $ENV == pr-${PR_NUM:-''} ]] ; then
+  aws ssm get-parameters --name /${PROJECT}/prs > $JSON_TEMPLATE # get all template envs
+  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_TEMPLATE ) ]] ; then
+    echo "Template PR variables /${PROJECT}/prs not found."
+    exit 1
+  fi
+  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_FULL ) ]] ; then # check if it's a new PR
+    echo "New PR - create variables."
+    jq -r '.Parameters | .[] | .Value' $JSON_TEMPLATE > $JSON
     if [[ -f herokles/set-custom-pr-envs.sh ]] ; then
-      source ./herokles/set-custom-pr-envs.sh
+      ./herokles/set-custom-pr-envs.sh $JSON
     fi
-    aws ssm put-parameter --type String --name /${PROJECT}/${ENV} --value "$JSON"
-  else
+    aws ssm put-parameter --type String --name /${PROJECT}/${ENV} --value "$( jq -c . $JSON )" # -c saves spaces
+  else # try to find new envs in /PROJECT/prs template, fill them in and push
+    update=$( mktemp )
+    jq -r '.Parameters | .[] | .Value' $JSON_FULL > $update
+    cat $update > $JSON
+    jq -r '.Parameters | .[] | .Value' $JSON_TEMPLATE > $update
+    cat $update > $JSON_TEMPLATE
+    echo > $update
```

Comment:
I would add a comment here that all the lines before are just to remove top level json junk from `json`and `json_template`. Together with `update` as var name it took me while to understand that this is not part of update logic but cleanup only :)

Reply:
It is a bit messy, I'll give you that :D I'll refactor it a bit.